### PR TITLE
In audit mode, suppress backtrace lines belonging to Chef itself.

### LINKED ISSUE #2754: disable stack trace when audits fail
When an audit fails I don't need to see the entire callstack; this just clutters the output.

Example output currently with the 12.2.0 alpha:

```
  2) check sshd configuration sshd configuration should force privilege separation
     Failure/Error: expect(config_file).to match(/^UsePrivilegeSeparation sandbox/)
       expected File "/etc/ssh/sshd_config" to match /^UsePrivilegeSeparation sandbox/
       Diff:
       @@ -1,2 +1,2 @@
       -/^UsePrivilegeSeparation sandbox/
       +File "/etc/ssh/sshd_config"

     # /var/chef/cache/cookbooks/openssh/recipes/audit_sshd.rb:46:in `block (3 levels) in from_file'
     # /opt/chef/embedded/apps/chef/lib/chef/audit/runner.rb:169:in `do_run'
     # /opt/chef/embedded/apps/chef/lib/chef/audit/runner.rb:33:in `run'
     # /opt/chef/embedded/apps/chef/lib/chef/client.rb:360:in `run_audits'
     # /opt/chef/embedded/apps/chef/lib/chef/client.rb:449:in `run'
     # /opt/chef/embedded/apps/chef/lib/chef/application.rb:261:in `block in fork_chef_client'
     # /opt/chef/embedded/apps/chef/lib/chef/application.rb:249:in `fork'
     # /opt/chef/embedded/apps/chef/lib/chef/application.rb:249:in `fork_chef_client'
     # /opt/chef/embedded/apps/chef/lib/chef/application.rb:215:in `block in run_chef_client'
     # /opt/chef/embedded/apps/chef/lib/chef/local_mode.rb:38:in `with_server_connectivity'
     # /opt/chef/embedded/apps/chef/lib/chef/application.rb:201:in `run_chef_client'
     # /opt/chef/embedded/apps/chef/lib/chef/application/client.rb:361:in `block in interval_run_chef_client'
     # /opt/chef/embedded/apps/chef/lib/chef/application/client.rb:351:in `loop'
     # /opt/chef/embedded/apps/chef/lib/chef/application/client.rb:351:in `interval_run_chef_client'
     # /opt/chef/embedded/apps/chef/lib/chef/application/client.rb:341:in `run_application'
     # /opt/chef/embedded/apps/chef/lib/chef/application.rb:58:in `run'
```

### PR DESCRIPTION
**Pull request by juliandunn**
Fixes #2754.

### CODE REVIEW

**Comment by jdmundrawala** [comment]
:+1: 
cc @tyler-ball @mcquin

File: lib/chef/audit/runner.rb [hunk 1]
```diff
@@ -96,6 +97,7 @@ def configure_rspec
         RSpec.configure do |c|
           c.color = Chef::Config[:color]
           c.expose_dsl_globally = false
+          c.backtrace_exclusion_patterns << Regexp.new(Chef::Config.platform_specific_path('/opt/chef'))
```
Comment (btm) [inline lib/chef/audit/runner.rb]:
```
irb(main):002:0> Chef::Config.platform_specific_path('/opt/chef')
=> "C:\\chef"
```

The Windows MSI still installs to C:\opscode\chef

**Comment by btm** [comment]
I pondered if there's any instance that you'd want to stacktrace for debugging purposes when something goes wrong (besides when a test fails as expected), but couldn't think of one.

I'd like to see equal support for Windows before merge, and at least a unit test like ```expect(RSpec.configuration.backtrace_exclusion_patterns).to include(/C:\opscode\chef/)```

File: lib/chef/audit/runner.rb (hunk 1 above)
Comment (tyler-ball) [inline lib/chef/audit/runner.rb]: I think you could introspect `__FILE__` and use the first couple path segments from that to setup the exclusion pattern

**Comment by juliandunn** [comment]
Thanks. What's the current wisdom to evaluate whether we're running on Windows, both in a code context and test context? I will update this PR.

**Comment by lamont-granquist** [comment]
there's the `lib/chef/platform_helper.rb` (i think) `#windows?` check

**Comment by juliandunn** [comment]
@btm I took a crack at writing a unit test for this, but with the `__FILE__` implementation as suggested by @tyler-ball, I can't see a decent way to do write one without actually just rewriting the same logic as in the code. Thoughts?

File: lib/chef/audit/runner.rb [hunk 2]
```diff
@@ -96,6 +97,8 @@ def configure_rspec
         RSpec.configure do |c|
           c.color = Chef::Config[:color]
           c.expose_dsl_globally = false
+          separator = File::ALT_SEPARATOR ? File::ALT_SEPARATOR : File::SEPARATOR
+          c.backtrace_exclusion_patterns << Regexp.new(__FILE__.split(separator)[0..2].join(separator))
```
Comment (lamont-granquist) [inline lib/chef/audit/runner.rb]: on windows isn't this going to give you `%w{c: opscode chef}` while on unix you'll get `%w{opt chef embedded}` and on my rvm install you'll get `%w{home lamont .rvm}`, etc?  You probably want a `-2` on the ending range or something like that?

**Comment by kmacgugan** [comment]
Hey all, I added a new pull request off my branch for this fix. https://github.com/chef/chef/pull/3001
It has a unit test that includes windows like paths.

File: lib/chef/audit/runner.rb (hunk 2 above)
Comment (tyler-ball) [inline lib/chef/audit/runner.rb]: I think we do want what you just described @lamont-granquist - on Windows we don't want to display any backtraces from `C:\opscode\chef\**\*`

**Comment by btm** [comment]
Fixed via #3001

**Pull request closed without merging**